src/ChangePrecision.jl

```diff
@@ -84,19 +84,21 @@ function changeprecision(T, x::Float64)
         return :(parse($T, $(string(x))))
     end
 end
-function changeprecision(T, ex::Expr)
-    if Meta.isexpr(ex, :call, 3) && ex.args[1] == :^ && ex.args[3] isa Int
+function changeprecision(T, ex::Expr, mod)
+    if Meta.isexpr(ex, :macrocall)
```

Comment:
Maybe we should just expand the `@.` macro, i.e. only expand a macrocall to `__dot__`?   To do that without expanding other nested macros, we would have to do something like:
```jl
if Meta.isexpr(ex, :macrocall) && ex.args[1] == Symbol("@__dot__")
    return changeprecision(T, Base.Broadcast.__dot__(ex.args[end]))
```
```

Reply:
That is more surgical, but things like `@..` (that DiffEq uses and that has a few more bells and whistles on top of `@.`) won't work. I'm fine with whatever.

Btw, is
```
elseif Meta.isexpr(ex, :call, 3) && ex.args[1] == :^ && ex.args[3] isa Int
else
```
a typo?

Reply:
whoops, looks like I was trying to do something special for literal powers but forgot to fill it in?

Reply:
Ultimately, it's hard to do something completely reliable here… ChangePrecision is meant for quick hacks and experiments, not for long-term usage.    I'd prefer to err on the side of narrower transformations, though, even if it fails noisily in some cases.

Reply:
OK, done